packages/diagrams/frontend/sirius-components-diagrams/src/renderer/selection/useDiagramSelection.ts

```diff
@@ -35,7 +35,14 @@ export const useDiagramSelection = (onShiftSelection: boolean): void => {
       .sort((id1: string, id2: string) => id1.localeCompare(id2));
     const selectedDiagramElementIds = [...getNodes(), ...getEdges()]
       .filter((element) => element.selected)
-      .map((element) => element.data?.targetObjectId ?? '')
+      .reduce((uniqueIds: string[], element) => {
+        const elementId: string = element.data?.targetObjectId ?? '';
+        const existingEntry = uniqueIds.find((id: string) => id === elementId);
+        if (!existingEntry) {
+          uniqueIds.push(elementId);
+        }
+        return uniqueIds;
+      }, [])
```

Comment:
It feels like using an intermediate `Set()` would be clearer? (and technically more performant as this is O(n²) with `find()` called on every element).

Reply:
Done